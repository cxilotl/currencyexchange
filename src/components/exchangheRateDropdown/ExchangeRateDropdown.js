import React from 'react';
import PropTypes from 'prop-types';
import cssStyles from './ExchangeRateDropdown.module.scss';

const ExchangeRateDropdown = ({ rates }) => {
  const id = `exchange-rate`;
  return(
    <div className={ cssStyles.wrapper }>
      <select
        id={ id }
        data-testid={ id }
        className={ cssStyles.select }
      >
        {
          rates.map((item) => (
              <option key={ item } value={ item } className={ cssStyles.option }>
                { item }
              </option>
          ))
        }
      </select>
    </div>
  );
};

ExchangeRateDropdown.propTypes = {
  rates: PropTypes.array
};

export default ExchangeRateDropdown;