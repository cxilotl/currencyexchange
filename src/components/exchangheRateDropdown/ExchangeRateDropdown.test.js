import React from 'react';
import renderer from 'react-test-renderer';
import ExchangeRateDropdown from '../exchangheRateDropdown/ExchangeRateDropdown';

describe('ExchangeRateDropdown', () => {
  const exchangeRates = [
    `£1 = €1.3444`,
    `€1 = £0.7438`
  ];
  it('Snapshot', () => {
    const component = renderer.create(
      <ExchangeRateDropdown rates={ exchangeRates }/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});