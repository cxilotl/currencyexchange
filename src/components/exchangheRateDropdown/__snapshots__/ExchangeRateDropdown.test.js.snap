// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRateDropdown Snapshot 1`] = `
<div
  className="wrapper"
>
  <select
    className="select"
    data-testid="exchange-rate"
    id="exchange-rate"
  >
    <option
      className="option"
      value="£1 = €1.3444"
    >
      £1 = €1.3444
    </option>
    <option
      className="option"
      value="€1 = £0.7438"
    >
      €1 = £0.7438
    </option>
  </select>
</div>
`;
